### ISSUE TITLE
LiDAR Distillation: Bridging the Beam-Induced Domain Gap for 3D Object Detection

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#13]
### Paper Link
https://arxiv.org/abs/2203.14956
### Code Link
https://github.com/weiyithu/LiDAR-Distillation
## Motivation

Beam-induced domain gap directly comes from the sensors.

<img width="600" alt="image" src="https://user-images.githubusercontent.com/50613939/202146187-79b80be5-2cb2-4051-9533-fe2f923db502.png">

<img width="600" alt="image" src="https://user-images.githubusercontent.com/50613939/202161013-32deca44-e25e-4d01-a734-96bc2454bb7d.png">

## Methodology

<img width="600" alt="image" src="https://user-images.githubusercontent.com/50613939/202146724-33aa6306-5773-463a-9398-04166a8f9410.png">

### Overall 
1. **Pseudo Low-beam Data Generation**: downsample high-beam data to pseudo low-beam point clouds.
2. **Distillation from High-beam Data**: employs a teacher-student framework, where teacher and student are inputted by different beams of training data, student network mimics the BEV feature maps predicted by the teacher model.
3. **Progressive Knowledge Transfer**: If the source domain is 64-beam and the target domain is 16-beam, we first need to generate pseudo 32-beam data and train a student model on it.

### Pseudo Low-beam Data Generation

<img width="600" alt="image" src="https://user-images.githubusercontent.com/50613939/202155529-4e78d924-6f9a-4e92-abf8-c68c3ca8bd88.png">

1. Transfer Cartesian coordinates (x, y, z) to spherical coordinates, where $\phi $ and $\theta$ are azimuth and zenith angles.

$$
\begin{aligned}
\theta &=\arctan \frac{z}{\sqrt{x^2+y^2}} \\
\phi &=\arcsin \frac{y}{\sqrt{x^2+y^2}}
\end{aligned}
$$

2. Apply K-Means algorithm on zenith angles \phi to find $B_s$ cluster centers as beam angles
3. Define the equivalent beams $B_t^{\prime}$ to the source domain as follows, where $\left[\alpha_s, \beta_s\right]$ and $\left[\alpha_t, \beta_t\right]$ represent the vertical field of view in source and target domains. With the beam label of each point, we can easily downsample $B_s$ beams. 

$$
B_t^{\prime}=\left[\frac{\alpha_s-\beta_s}{\alpha_t-\beta_t} B_t\right]
$$

### Distillation from High-beam Data

- Mimic the BEV features sampled from the region of interest (ROI)
- The loss function that the student network is: 

$$
\begin{gathered}
L=L_{g t}+\lambda L_m^{3 D} \\
L_m^{3 D}=\frac{1}{N} \sum_i \frac{1}{M_i} \sum_j\left\|b_t^{i j}-b_s^{i j}\right\|_2
\end{gathered}
$$

- $b^{i j}$ is the $j$ th ROI of the $i$ th sample's BEV feature maps. 
- $N$ and $M_i$ represent sample number and ROI number in the $i$ th sample. 
- $L_{g t}$ is the original objective function for $3 \mathrm{D}$ object detection.
- $\lambda$ is the mimic loss weight. 

### Progressive Knowledge Transfer
If the source domain is 64-beam and the target domain is 16-beam, we first need to generate pseudo 32-beam data and train a student model on it. This student model will be used as the teacher network in the next iteration.

## Implementation 

- 3D detectors: PointPillars, SECOND-IoU and PV-RCNN.
- Adopt same data augmentation as ST3D 

## Experiments 

<img width="700" alt="image" src="https://user-images.githubusercontent.com/50613939/202170063-9b818462-cc38-4e5b-9ea2-c452a44ed590.png">

<img width="700" alt="image" src="https://user-images.githubusercontent.com/50613939/202170109-e74d1cf8-c394-44ae-b670-0b8b9a1a9910.png">

- Build the synthetic low-beam target domain by uniformly downsampling the validation set of KITTI.
- UDA method ST3D will degrade the performance. This is mainly due to different point cloud densities in source and target domains, which are not friendly to the self-training framework.
